src/main/java/cloud/eppo/ConfigurationManager.java

```diff
@@ -0,0 +1,244 @@
+package cloud.eppo;
+
+import cloud.eppo.api.Configuration;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConfigurationManager {
```

Comment:
add a 1-2 paragraph comment describing the purpose of this class. also seems like this shouldn't be public (i don't see any public methods)